pkg/distributor/distributorerror/error.go

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package distributorerror
+
+import (
+	"fmt"
+
+	"github.com/grafana/mimir/pkg/util/globalerror"
+	"github.com/grafana/mimir/pkg/util/validation"
+)
+
+type distributorPushError struct {
+	err error
+}
+
+// Error makes distributorPushError implement error interface.
+func (e distributorPushError) Error() string {
+	return e.err.Error()
+}
+
+// Unwrap makes distributorPushError implement Wrapper interface.
+func (e distributorPushError) Unwrap() error {
+	return e.err
+}
+
+// ReplicasNotMatchError is an implementation of Error,
+// meaning that replicas do not match.
+type ReplicasNotMatchError struct {
+	replica, elected string
+}
+
+func NewReplicasNotMatchError(replica, elected string) ReplicasNotMatchError {
+	return ReplicasNotMatchError{
+		replica: replica,
+		elected: elected,
+	}
+}
+
+// Error makes ReplicasNotMatchError implement error interface.
+func (e ReplicasNotMatchError) Error() string {
+	return fmt.Sprintf("replicas did not mach, rejecting sample: replica=%s, elected=%s", e.replica, e.elected)
+}
+
+// TooManyClustersError is an implementation of Error,
+// meaning that there are too many HA clusters (globalerror.TooManyHAClusters).
+type TooManyClustersError struct {
+	limit int
+}
+
+func NewTooManyClustersError(limit int) TooManyClustersError {
+	return TooManyClustersError{
+		limit: limit,
+	}
+}
+
+// Error makes TooManyClustersError implement error interface.
+func (e TooManyClustersError) Error() string {
+	return globalerror.TooManyHAClusters.MessageWithPerTenantLimitConfig(
+		fmt.Sprintf("the write request has been rejected because the maximum number of high-availability (HA) clusters has been reached for this tenant (limit: %d)", e.limit),
+		validation.HATrackerMaxClustersFlag)
+}
+
+// ValidationError is an implementation of Error,
+// used to represent all implementations of validation.ValidationError.
+type ValidationError struct {
+	distributorPushError
+	err string
+}
+
+func NewValidationError(err validation.ValidationError) ValidationError {
+	return ValidationError{
+		distributorPushError: distributorPushError{
+			err: err,
+		},
+		err: err.Error(),
+	}
+}
+
+func (e ValidationError) Error() string {
+	return e.err
+}
+
+// IngestionRateError is an implementation of Error,
+// used to represent the ingestion rate limited error (globalerror.IngestionRateLimited).
```

Comment:
Can we document somewhere what is the relation between `distributorerror` and `globalerror` packages? 

This package wraps several error values from that package with specific types here. Where do those live? I think the fact that those are both constants and error doesn't really help.